var/spack/repos/builtin/packages/py-numcodecs/package.py

```diff
@@ -21,22 +21,38 @@ class PyNumcodecs(PythonPackage):
     import_modules = ["numcodecs"]
 
     version("master", branch="master", submodules=True)
+    version("0.11.0", sha256="6c058b321de84a1729299b0eae4d652b2e48ea1ca7f9df0da65cb13470e635eb")
     version("0.7.3", sha256="022b12ad83eb623ec53f154859d49f6ec43b15c36052fa864eaf2d9ee786dd85")
     version("0.6.4", sha256="ef4843d5db4d074e607e9b85156835c10d006afc10e175bda62ff5412fca6e4d")
 
     variant("msgpack", default=False, description="Codec to encode data as msgpacked bytes.")
 
+    depends_on("python@3.8:", when="@0.11:", type=("build", "link", "run"))
     depends_on("python@3.6:3", when="@0.7:", type=("build", "link", "run"))
```

Comment:
```suggestion
    depends_on("python@3.6:3", when="@0.7:0.10", type=("build", "link", "run"))
```
Otherwise upper bound will also apply to latest version

Reply:
I was aware of that, probably all python packages should have `@:3`

Reply:
Probably, not excited for that transition. Until then, I usually just follow what the devs document.